build.gradle

```diff
@@ -78,13 +74,21 @@ task clean(type: Delete) {
     delete rootProject.buildDir
 }
 
-detekt {
-    profile("main") {
-        input = "$projectDir"
-        config = "$project.rootDir/detekt-config.yml"
-        filters = ".*test.*,.*/resources/.*,.*/tmp/.*"
-        output = "${project.buildDir}/reports/detekt/"
-        parallel = true
-        disableDefaultRuleSets = false
-    }
+configurations {
+    detekt
+ }
+
+dependencies {
+    detekt "io.gitlab.arturbosch.detekt:detekt-cli:1.12.0"
```

Comment:
Why do we switch to use detekt-cli? 🤔 
Also switch from using `detekt` block to a custom task?

Reply:
creating a task is easier so you can still run `./gradlew detekt` but also you can run it directly from the gradle file, it looks like this (just like task :clean we had before):
![image](https://user-images.githubusercontent.com/13435717/94658456-0491ae80-032d-11eb-8bc0-088f8cb5ca3e.png)

Reply:
I'll recheck whether `detekt-cli` is necessary 🤔  from a setup tutorial I think they used it and I didn't realize that 🤔

Reply:
updated 🙏